plc4j/api/src/main/java/org/apache/plc4x/java/api/messages/PlcFieldRequest.java

```diff
@@ -20,11 +20,13 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 import org.apache.plc4x.java.api.model.PlcField;
 
-import java.util.Collection;
+import java.util.LinkedHashSet;
 
 public interface PlcFieldRequest extends PlcRequest {
 
-    Collection<String> getFieldNames();
+    int getNumFields();
+
+    LinkedHashSet<String> getFieldNames();
```

Comment:
Because that is the most generic container that is: 
- A Map (We need to map names to obects)
- Preserves the order (might be obsolete now ... have to re-check)